pages/index.js

```diff
@@ -19,6 +20,9 @@ export default function Home({ dogs }) {
       <Link href="/newdog">
```

Comment:
Placing buttons inside Links is somehow corrupting the `Link` functionality. So better create e.g. a `StyledLink` component which can get the same styles as your button component: 
```js 
styled(Link)`
  display: flex;
  margin: auto;
  margin-bottom: 25px;
  background-color: #445540;
  color: white;
  padding: 5px 30px;
  font-size: 16px;
  border-radius: 15px;
`;
```
This way you don't need to use buttons and you can use the same `StyledLink` component for every Link without wrapping it around a button:

```js
<StyledLink href="/newdog">Add Dog</Link>
```